test/integration/features/profile.feature

```diff
@@ -2,104 +2,229 @@
+     And stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift start" succeeds
       Then stdout should contain
       """
-      Profile 'minishift' set as active profile.
+      Add-on 'registry-route' created docker-registry route.
       """
-Scenario: User should not be able to create profile with blank profile name
-    When executing "minishift profile set"
-     Then exitcode should equal "1"
-     Then stderr should contain
+      And profile "foo" should have state "Running"
+  
+  @minishift-only
+  Scenario: As user, I can list add-ons in profile 'foo' 
+    Given profile "foo" has state "Running"
+     And executing "minishift profile set foo" succeeds
+     And stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift addons list"
+     Then stdout should match "admin-user\s*: disabled\s*P\(0\)"
+      And stdout should match "anyuid\s*: disabled\s*P\(0\)"
+      And stdout should match "registry-route\s*: enabled\s*P\(0\)"
+      And stdout should match "xpaas\s*: disabled\s*P\(0\)"
+
+  Scenario: Checking that memory value was applied on profile 'foo'
+    When executing "minishift ssh -- less /proc/meminfo" succeeds
+     Then stdout should match "MemTotal:\s*4[0-1][0-9]{5}\s*kB"
+
+  Scenario: Checking that disk-size value was applied on profile 'foo'
+    When executing "minishift ssh -- sudo fdisk -l | grep Disk" succeeds
+     Then stdout should match "Disk \/dev\/sda: 3[0-2]\.?[0-9]{0,2} (GB|GiB)"
+
+  Scenario: Checking that cpus value was applied on profile 'foo'
+    When executing "minishift ssh -- cat /proc/cpuinfo" succeeds
+     Then stdout should match "processor\s*: 0"
+      And stdout should match "processor\s*: 1"
+      And stdout should match "processor\s*: 2"
+      And stdout should match "processor\s*: 3"
+      And stdout should not match "processor\s*: [4-9]"
+
+  Scenario: As user, I can execute command on non active running profile 'minishift' from active profile 'foo'
+    Given profile "foo" has state "Running"
+     And executing "minishift profile set foo" succeeds
+     And stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift --profile minishift stop" succeeds
+     Then stdout should contain
+     """
+     Stopping local OpenShift cluster...
+     Cluster stopped.
      """
-     A profile name must be provided. Run `minishift profile list` for a list of existing profiles.
+     And executing "minishift --profile minishift status" succeeds
+      Then stdout should match "Minishift:\s*Stopped\nProfile:\s*minishift\nOpenShift:\s*Stopped"
+
+  Scenario Outline: As user, I can delete all created profiles
+    When executing "minishift profile delete <profilename> --force" succeeds
+     Then stdout should contain
      """
-    When executing "minishift profile set '  '"
+     Profile '<profilename>' deleted successfully
+     """
+  
+  Examples: Profile name
```

Comment:
This could be shortened I think, there is some repetition, for example `00_XYZ-` covers quite a lot. With `00_xYz-` it is even better. I would also suggest to use something like that, which is more complicated, instead of `foo` in all other cases. We can use something which looks like somebody is actually using, for example`small_VM-2` or `python_VM-1`.

What I like here is `foo-`, `profile_`, I am curious how is it with `_profile` or `-profile`.

Reply:
-profile won't be a valid usecase. ```minishift profile``` interprets this as flag. We can take some better permutation/combination using the separator ```_``` and ```-```. For more details i would suggest to have a look on #1749

Reply:
@amitkrout sure, but can that be passed when quotes are used? that should be possible AFAIU. But the discussion is ongoing on this: https://github.com/minishift/minishift/pull/1749#issuecomment-347812553

Reply:
> but can that be passed when quotes are used? that should be possible AFAIU

Just saw the question. No it should not work. Interpreter still consider this as a flag